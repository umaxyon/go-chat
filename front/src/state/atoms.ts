import { atom } from "recoil"

export type Member = {
    user: string
}

export type FeedRow = {
    id: string,
    MessageType: "comment" | "addMember" | "leaveMember" | "system",
    user: string,
    text: string,
    createdAt: any
}

export type LoginData = {
    user: string,
    token: string,
    disconnect?: () => void
}

export const membersState = atom<Member[]>({
    key: "members",
    default: [],
})

export const commentsState = atom<FeedRow[]>({
    key: "comments",
    default: [],
})

export const loginState = atom<LoginData>({
    key: "loginUser",
    default: undefined,
})