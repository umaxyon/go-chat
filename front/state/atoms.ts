import { atom } from "recoil"

export type Member = {
    user: string
}

export type Comment = {
    id: string,
    user: string,
    text: string,
    createdAt: any
}

export type LoginData = {
    user: string,
    disconnect?: () => void
}

export const membersState = atom<Member[]>({
    key: "members",
    default: [],
})

export const commentsState = atom<Comment[]>({
    key: "comments",
    default: [],
})

export const loginState = atom<LoginData>({
    key: "loginUser",
    default: undefined,
})