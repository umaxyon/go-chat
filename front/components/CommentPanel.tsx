import React from 'react'
import { Comment } from '../state/atoms'


type CommentPanelProps = {
    comment: Comment
}

const CommentPanel: React.FC<CommentPanelProps> = ({ comment }) => {
    return (
        <div className="w-full p-2 bg-green-100 text-sm">
            {comment.text}
        </div>
    )
}
export default CommentPanel
