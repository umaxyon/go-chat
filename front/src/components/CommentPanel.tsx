import React from 'react'
import { formatYMDHms } from 'src/utils/utils'
import { Comment } from '../state/atoms'


type CommentPanelProps = {
    comment: Comment
}

const CommentPanel: React.FC<CommentPanelProps> = ({ comment }) => {
    const createdAt = formatYMDHms(new Date(comment.createdAt))

    return (
        <div className="w-full pt-2 text-xs">
            <div className="flex flex-row w-full justify-between pb-1">
                <div className="inline-block py-1 px-2.5 leading-none text-center whitespace-nowrap align-baseline font-bold bg-gray-200 text-gray-700 rounded-full">
                    {comment.user}
                </div>
                <div className="text-slate-400">
                    {createdAt}
                </div>
            </div>
            <div className="ml-7 p-2 bg-green-200 text-sm balloon rounded-lg">
                {comment.text}
            </div>
        </div>
    )
}
export default CommentPanel
