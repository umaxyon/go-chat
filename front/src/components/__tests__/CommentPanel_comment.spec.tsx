import CommentPanel from "@/components/CommentPanel"
import "@testing-library/jest-dom/extend-expect";
import { cleanup, render, screen } from '@testing-library/react';
import renderer from "react-test-renderer"
import { FeedRow } from "../../state/atoms";

afterEach(() => cleanup())

const TEST_COMMENT: FeedRow = {
    id:   "1",
    MessageType: "comment",
    user: "test_user",
    text: "あいうえお",
    createdAt: "2022-07-09T22:04:00.2890199+09:00"
}

const sut = (<CommentPanel feed={TEST_COMMENT}/>)

describe('CommentPanelテスト(comment)', () => {
    it('snapshot test', () => {
        const component = renderer.create(sut);
        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('commentタイプの場合はtextが表示されること', () => {
        render(sut)
        expect(screen.getByText(`${TEST_COMMENT.text}`)).toBeInTheDocument()
    })
})
