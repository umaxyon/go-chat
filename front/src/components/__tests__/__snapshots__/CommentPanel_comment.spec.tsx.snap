// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentPanelテスト(comment) snapshot test 1`] = `
<div
  className="w-full pt-2 text-xs"
>
  <div
    className="flex flex-row w-full justify-between pb-1"
  >
    <div
      className="inline-block py-1 px-2.5 leading-none text-center whitespace-nowrap align-baseline font-bold bg-gray-200 text-gray-700 rounded-full"
    >
      test_user
    </div>
    <div
      className="text-slate-400"
    >
      2022/07/09 22:04
    </div>
  </div>
  <div
    className="ml-7 p-2 bg-green-200 text-sm balloon rounded-lg"
  >
    あいうえお
  </div>
</div>
`;
